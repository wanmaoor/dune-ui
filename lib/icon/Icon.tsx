import React from "react";
import classNames from "classnames";
import './importAll'
import './index.less'

interface IIconProps extends React.SVGAttributes<SVGElement>{
  name: string
}

const Icon: React.FunctionComponent<IIconProps> = ({ name, className ,...restProps}) => {
  return (<svg className={classNames('icon', className)} {...restProps}>
    <use xlinkHref={`#${name}`}/>
  </svg>)

}

export default Icon
