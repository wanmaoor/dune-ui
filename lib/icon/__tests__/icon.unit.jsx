import React from "react";
import renderer from "react-test-renderer";
import Icon from '../Icon'
import {mount} from "enzyme";
describe('Icon组件测试', () => {
  it('should render success', function () {
    const json = renderer.create(<Icon name={'alipay'} />).toJSON()
    expect(json).toMatchSnapshot()
  });

  it('should click success', function () {
    const fn = jest.fn()
    const component = mount(<Icon name={'alipay'} onClick={fn} />)
    component.find('svg').simulate('click')
    expect(fn).toBeCalled()
  });
})
