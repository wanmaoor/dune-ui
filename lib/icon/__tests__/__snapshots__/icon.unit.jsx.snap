// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon组件测试 should render success 1`] = `
<svg
  className="icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
